worker/rpc_task.go

```diff
@@ -122,11 +159,47 @@ func (r *RPCTask) AppendExecutorStdout(i int, s string) {
 
 // AppendExecutorStderr appends to an executor's stderr log.
 func (r *RPCTask) AppendExecutorStderr(i int, s string) {
-	r.client.UpdateExecutorLogs(&pbf.UpdateExecutorLogsRequest{
+	r.updateExecutorLogs(&pbf.UpdateExecutorLogsRequest{
 		Id:   r.taskID,
 		Step: int64(i),
 		Log: &tes.ExecutorLog{
 			Stderr: s,
 		},
 	})
 }
+
+func (r *RPCTask) updateWorker(req *pbf.Worker) (*pbf.UpdateWorkerResponse, error) {
+	ctx, cleanup := context.WithTimeout(context.Background(), r.updateTimeout)
+	resp, err := r.client.UpdateWorker(ctx, req)
+	cleanup()
+	return resp, err
+}
+
+func (r *RPCTask) updateExecutorLogs(up *pbf.UpdateExecutorLogsRequest) error {
+	ctx, cleanup := context.WithTimeout(context.Background(), r.updateTimeout)
+	_, err := r.client.UpdateExecutorLogs(ctx, up)
+	cleanup()
+	return err
+}
+
+func (r *RPCTask) updateTaskLogs(up *pbf.UpdateTaskLogsRequest) error {
+	ctx, cleanup := context.WithTimeout(context.Background(), r.updateTimeout)
+	_, err := r.client.UpdateTaskLogs(ctx, up)
+	cleanup()
+	return err
+}
+
+type taskClient struct {
+	tes.TaskServiceClient
+	pbf.SchedulerServiceClient
+}
+
+func newTaskClient(conf config.Worker) (*taskClient, error) {
+	conn, err := grpc.Dial(conf.ServerAddress, grpc.WithInsecure())
```

Comment:
Should we have a timeout on this and/or should this block until the connection is made?